packages/authentication/cardstack/middleware.js

```diff
@@ -143,6 +128,24 @@ class Authentication {
       user = await this._processExternalUser(result, source);
     }
 
+    if (result && result.meta && result.meta.partialSession) {
+      if (!user) {
+        // top-level meta is not passed through (it was for
+        // communicating from plugin to us). Plugins could use
+        // resource-level metadata instead if they want to.
+        delete result.meta;
+        user = result;
+      }
+
+      if (user.data.type !== 'partial-sessions') {
+        user.data.type = 'partial-sessions';
+      }
```

Comment:
This was originally intended as a default, not an override. Users can have their own model for partial sessions, we only care about meta.partialSession.

Reply:
Sure